packages/app-desktop/integration-tests/util/mockNextShowMessageDialog.ts

```diff
@@ -0,0 +1,84 @@
+
+import type { ElectronApplication } from '@playwright/test';
+import { BrowserWindow, MessageBoxOptions } from 'electron';
+
+const mockNextShowMessageCall = (
+	electronApp: ElectronApplication,
+
+	// Only matches messages that match
+	messagePattern: RegExp,
+
+	// Chooses the answer that matches this pattern
+	answerPattern: RegExp,
+) => {
+	return electronApp.evaluate(({ BrowserWindow, dialog }, [messagePattern, answerPattern]) => {
+		// Mock both showMessageBox and showMessageBoxSync. The app should be able to switch between
+		// the two without affecting the tests.
+		const mockDialogMethod = (methodName: 'showMessageBox'|'showMessageBoxSync') => {
+			let originalShowMessageBox = dialog[methodName];
+
+			dialog[methodName] = (
+				optionsArgOrWindow: BrowserWindow|MessageBoxOptions, optionsArg?: MessageBoxOptions,
+			) => {
+				let options: MessageBoxOptions;
+				if (optionsArgOrWindow instanceof BrowserWindow) {
+					options = optionsArg!;
+				} else {
+					options = optionsArgOrWindow;
+				}
+
+				let result: any;
+
+				if (options.message.match(messagePattern)) {
+					dialog[methodName] = originalShowMessageBox as any;
+
+					const buttons = options.buttons ?? ['OK'];
+					let returnIndex = -1;
+					for (let i = 0; i < buttons.length; i++) {
+						if (buttons[i].match(answerPattern)) {
+							returnIndex = i;
+							break;
+						}
+					}
+
+					if (returnIndex === -1) {
+						throw new Error(`Unable to find button matching ${answerPattern}`);
+					}
+
+					result = {
+						response: returnIndex,
+						checkboxChecked: false,
+					};
+				} else {
```

Comment:
For the new tests, it should be sufficient to mock *all* `showMessage` calls that match a pattern, rather than just the next. This change should allow much of this function to be simplified, as much of it is related to restoring the original `dialog.showMessageBox` function.

Reply:
Done in e4f63faeb06eaf2e9806dcbc570ad2f5840acf5e. Note that the original might be more useful in some cases, so it may make sense to revert this.